File: backend/shop-floor-assistance-backend/src/main/java/com/shopfloor/backend/api/transferobjects/OrderTO.java
package com.shopfloor.backend.api.transferobjects;

import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Getter
@Setter
public class OrderTO {

    private Long id;

    private String orderNumber;

    private String name;

    private String description;

    private List<WorkflowTO> workflows;

    private Long createdBy;

    private Long updatedBy;

    private Date createdAt;

    private Date updatedAt;

    public OrderTO() {
        this.workflows = new ArrayList<WorkflowTO>();
    }

}

File: frontend/shop-floor-assistance-frontend/src/app/types/orderTO.ts
import { workflowTO } from "./workflowTO";
import { productTO } from "./productTO"; 
import { equipmentTO } from "./equipmentTO"; 

export type orderTO = {
    id?: number; 
    orderNumber: string;
    name: string;
    description?: string; 

    workflows: workflowTO[]; 
    
    createdBy?: number;
    updatedBy?: number;
    createdAt?: Date;
    updatedAt?: Date;
    productBefore: productTO; 
    productAfter: productTO;
    equipment: equipmentTO[];
    total_time_required?: number;
};
